src/components/UserDashboard/CavaticaProjects/index.js

```diff
@@ -50,51 +44,53 @@ const CavaticaProjects = compose(injectState)(({ state: { integrationTokens } })
     });
   };
 
-  const unsetBadge = cardState => d => {
-    cardState.setBadge(null);
-  };
-
-  const tabToCreate = cardState => d => {
-    cardState.setIndex(1);
+  const tabToCreate = cardProps => d => {
+    cardProps.setIndex(1);
   };
 
   return (
-    <DashboardMulticard
-      inactive={!isConnected}
-      scrollable={isConnected}
-      tabs={
-        isConnected
-          ? [
-              {
-                title: 'Cavatica Projects',
-                nav: 'Projects',
-                component: cardState => (
-                  <CavaticaProvider onData={onCavaticaData(cardState)}>
-                    {({ projects, loading }) => (
-                      <Connected
-                        tabToCreate={tabToCreate(cardState)}
-                        projects={projects}
-                        loading={loading}
-                      />
-                    )}
-                  </CavaticaProvider>
-                ),
-              },
-              {
-                title: 'Create a Cavatica Project',
-                nav: 'Create',
-                component: cardState => (
-                  <Create
-                    onInit={unsetBadge(cardState)}
-                    onProjectCreated={onProjectCreationComplete(cardState)}
-                    onProjectCreationCancelled={onProjectCreationCanceled(cardState)}
-                  />
-                ),
-              },
-            ]
-          : [{ title: 'CAVATICA Projects', component: cardState => <NotConnected /> }]
-      }
-    />
+    <CavaticaProvider>
```

Comment:
@hlminh2000 Issue here. All tab content is loaded first time with the slider. When a user creates a new cavatica project we want the `CavaticaProvider` to refresh. Any thoughts on best approach?

Maybe have a `refresh` state at the `CavaticaProjects` level and pass it to `CavaticaProvider`

= `Create` Card =
`onProjectCreationComplete` function can toggle it from the `Create` tab content

= `CavaticaProvider` =
`componentShouldUpdate` if `refresh === true` get data  again

^ Seems a little convoluted

Reply:
I'd suggest a solution like the one in gen3's `Gen3UserProvider`. Basically the provider injects a `refresh` function which the consumer can call after its done with side effects :)